src/test/java/uk/nhs/hee/tis/trainee/forms/service/FormRPartAServiceImplTest.java

```diff
@@ -321,4 +332,41 @@ void shouldGetDraftFormRPartAById() {
     assertThat("Unexpected forename.", dto.getForename(), is(DEFAULT_FORENAME));
     assertThat("Unexpected surname.", dto.getSurname(), is(DEFAULT_SURNAME));
   }
+
+  @Test
+  void shouldPartialDeleteFormRPartAById() {
+    when(repositoryMock.findByIdAndTraineeTisId(DEFAULT_ID, DEFAULT_TRAINEE_TIS_ID))
+        .thenReturn(Optional.of(entity));
+
+    service.partialDeleteFormRPartAById(
+        DEFAULT_ID_STRING, DEFAULT_TRAINEE_TIS_ID, FIXED_FIELDS);
+
+    verify(repositoryMock).save(formRPartACaptor.capture());
+    FormRPartA dto = formRPartACaptor.getValue();
+    assertThat("Unexpected form ID.", dto.getId(), is(DEFAULT_ID));
+    assertThat("Unexpected trainee ID.", dto.getTraineeTisId(), is(DEFAULT_TRAINEE_TIS_ID));
+    assertThat("Unexpected lifecycle states.",
+        dto.getLifecycleState(), is(LifecycleState.DELETED));
+    assertThat("Unexpected forename.", dto.getForename(), is(nullValue()));
+    assertThat("Unexpected surname.", dto.getSurname(), is(nullValue()));
+  }
```

Comment:
I generally prefer individual field checks (like you've done) over building an "expectedResult" instance of the object being tested, but in this case it might be an appropriate way to test all other fields get nulled without having to test each and every field.

e.g.
```java
FormRPartA expectedDto = new FormRPartA();
expectedDto.setId(DEFAULT_ID);
expectedDto.setTraineeTisId(DEFAULT_TRAINEE_TIS_ID);
expectedDto.setLifeCycle(LifecycleState.DELETED);

assertThat("Unexpected DTO.", dto, is(expectedDto));
```